lte/gateway/c/core/oai/lib/n11/M5GSUCIRegistrationServiceClient.cpp

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright 2020 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "lte/gateway/c/core/oai/lib/n11/M5GSUCIRegistrationServiceClient.h"
+
+#include <cstring>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <thread>
+#include <cassert>
+
+#include "lte/gateway/c/core/oai/lib/3gpp/3gpp_38.413.h"
+#include "lte/gateway/c/core/oai/common/log.h"
+
+#include <google/protobuf/util/time_util.h>
+#include <grpcpp/impl/codegen/client_context.h>
+#include <grpcpp/impl/codegen/status.h>
+
+#include "lte/protos/subscriberdb.grpc.pb.h"
+#include "lte/protos/subscriberdb.pb.h"
+#include "orc8r/gateway/c/common/service_registry/includes/ServiceRegistrySingleton.h"
+#include "lte/gateway/c/core/oai/lib/n11/amf_client_proto_msg_to_itti_msg.h"
+
+using grpc::Channel;
+using grpc::ClientContext;
+using grpc::InsecureChannelCredentials;
+using grpc::Status;
+using magma::AsyncLocalResponse;
+using magma::ServiceRegistrySingleton;
+using magma::lte::M5GSUCIRegistrationAnswer;
+using magma::lte::M5GSUCIRegistrationRequest;
+
+extern task_zmq_ctx_t grpc_service_task_zmq_ctx;
+
+static void handle_decrypted_imsi_info_ans(
+    grpc::Status status, magma::lte::M5GSUCIRegistrationAnswer response,
+    amf_ue_ngap_id_t ue_id) {
+  MessageDef* message_p;
+
+  if (!status.ok()) {
+    std::cout << "get_decrypt_imsi_info fails with code " << status.error_code()
+              << ", msg: " << status.error_message() << std::endl;
+
+    return;
+  }
+
+  message_p =
+      itti_alloc_new_message(TASK_GRPC_SERVICE, AMF_APP_DECRYPT_IMSI_INFO_RESP);
+
+  itti_amf_decrypted_imsi_info_ans_t* amf_app_decrypted_imsi_info_resp;
+  amf_app_decrypted_imsi_info_resp =
+      &message_p->ittiMsg.amf_app_decrypt_info_resp;
+  memset(
+      amf_app_decrypted_imsi_info_resp, 0,
+      sizeof(itti_amf_decrypted_imsi_info_ans_t));

```

Comment:
this initialization is already done as part of `itti_alloc_new_message` on line 58.

Reply:
will avoid reinitialization of the object.

Reply:
Done